jupyter/swan.py

```diff
@@ -0,0 +1,560 @@
+"""
+ Copyright (c) 2017 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+
+This module contains the convince class to read experiment data and generate sensitivity
+profiles. See profile.py for more information.
+"""
+
+import sys
+import re
+import os
+import datetime
+import pickle
+from functools import partial
+import pandas as pd
+import numpy as np
+
+
+# -------------------------------
+# dataframe on disk caching layer
+# -------------------------------
+class DataFrameToCSVCache:
+    """ CSV based cache to store dataframe as compressed CSV."""
+
+    CACHE_DIR = '.experiments_csv_cache'
+
+    def _filename(self, experiment_id):
+        return os.path.join(self.CACHE_DIR, '%s.cache' % experiment_id)
+
+    def __contains__(self, experiment_id):
+        return os.path.exists(self._filename(experiment_id))
+
+    def __setitem__(self, experiment_id, df):
+        if not os.path.exists(self.CACHE_DIR):
+            os.makedirs(self.CACHE_DIR)
+        df.to_csv(self._filename(experiment_id), compression='bz2')
+
+    def __getitem__(self, experiment_id):
+        if experiment_id in self:
+            return pd.read_csv(self._filename(experiment_id), compression='bz2')
+        else:
+            raise KeyError()
+
+    def __call__(self, fun):
```

Comment:
@ppalucki - as a python newbe I would appreciate renaming `fun` so its name is more meaningful.